lib/kubeclient/common.rb

```diff
@@ -63,6 +70,9 @@ def initialize_client(
       @ssl_options = ssl_options
       @auth_options = auth_options
       @socket_options = socket_options
+      # Allow passing partial timeouts hash, without unspecified
+      # @timeouts[:foo] == nil resulting in infinite timeout.
```

Comment:
if you pass just one, `timeouts:{open: 10}` or `timeouts: {read: 20}`.  merge ensures `@timeouts` always has both.